plugins/chat/app/models/chat/channel.rb

```diff
@@ -107,6 +107,19 @@ def self.ensure_consistency!
       update_counts
     end
 
+    def can_be_joined_by?(user)
```

Comment:
shouldn't this be a guardian?

Reply:
It probably should, I think it should be something like `user.can_join_chat_channel?` but the thing is we already have such method, we call it here in implementation:

    def can_be_joined_by?(user)
      guardian = Guardian.new(user)
      guardian.can_chat? && guardian.can_join_chat_channel?(self)
    end

I think we should refactor the guardian like this: https://github.com/discourse/discourse/pull/21812.

Someone who cannot chat is also not able to join chat channels.

Reply:
https://github.com/discourse/discourse/pull/21812 is merged now, so we don't need this method anymore and I've removed it.

We can use `guardian.can_join_chat_channel?` when needed instead.